src/main/groovy/edu/oregonstate/mist/osuevents/resources/EventsResource.groovy

```diff
@@ -67,13 +68,51 @@ class EventsResource extends Resource {
  */
     @GET
     @Produces(MediaType.APPLICATION_JSON)
+    //TODO Add CSV MediaType to response object
     public Response getEvents(@Auth AuthenticatedUser _,
                               @QueryParam('format') String format) {
         def events = eventsDAO.getEvents()
-//@TODO add methods for returning csv and ics formats
-//        else if (format == "csv") {
-//
-//        } else if (format == "ics") {
+        def resultObject
+
+        if (!format) {
+            events.each {
+                resultObject.data += new ResourceObject(
+                        id: it.eventID,
+                        type: 'event',
+                        attributes: it
+                )
+            }
+        } else if (format == "csv") {
+            resultObject = getResultObject(events)
+
+            def time = new Date().time
+            String csvfilename = "events_gospel" + time + ".csv"
+
+            CSVWriter writer = new CSVWriter(new FileWriter(csvfilename))
+
+            //CSV Layout Header
+            writer.writeNext(["Title" , "Description" , "Date From" , "Date To" , "Recurrence" ,
+                              "Start Time", "End Time" , "Location" , "Address" , "City" , "State" ,
+                              "Event Website" , "Room" , "Keywords" , "Tags" , "Photo URL" ,
+                              "Ticket URL" , "Cost" , "Hashtag" , "Facebook URL" , "Group" ,
+                              "Department" , "Allow User Activity" , "Allow User Attendance" ,
+                              "Visibility" , "Featured Tabs" , "Sponsored" , "Venue Page Only" ,
+                              "Exclude From Trending" , "Event Types" , "Top level filter" ,
```

Comment:
We don't want to hard code any filters or custom fields. These things will differ depending on what we have stored in our local cache. Localist expects each custom field and filter to be it's own column. You can use [this DAO method ](https://github.com/osu-mist/osu-events-api/blob/develop/src/main/groovy/edu/oregonstate/mist/osuevents/db/EventsDAO.groovy#L276-L278) to get all custom fields from the local cache and [this DAO method](https://github.com/osu-mist/osu-events-api/blob/develop/src/main/groovy/edu/oregonstate/mist/osuevents/db/EventsDAO.groovy#L340-L342) to get filters from the local cache.